### PR TITLE
feat(thirdweb): enables oauth redirect flow

### PR DESCRIPTION
**Pull request by gregfromstl**
### TL;DR

This PR adds support for redirecting the user to a new page for OAuth logins instead of using a pop-up window.

### What changed?

1. Added the `socialLoginRedirect` prop to various components and functions across the codebase.
2. Implemented the `loginWithOauthRedirect` function to facilitate OAuth login redirections.
3. Updated the default domain URLs to use `thirdweb-dev.com`.
4. Refactored existing OAuth login code to support the new `socialLoginRedirect` functionality.

### How to test?

1. Set `socialLoginRedirect` to `true` in the desired component or function.
2. Trigger an OAuth login and verify that it redirects to a new page instead of opening a pop-up.

### Why make this change?

This change provides a better user experience for OAuth logins by avoiding pop-up blockers and providing a smoother, more reliable login process.

---

---

## PR-Codex overview
This PR introduces a new `socialLoginMode` option for OAuth login page behavior in various components.

### Detailed summary
- Added `socialLoginMode` option with values "popup" or "redirect" to control OAuth login page behavior
- Updated multiple components to include and utilize the `socialLoginMode` option
- Deprecated `redirectUrl` in favor of `socialLoginMode` in `InAppWalletConnectionOptions`

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/wallets/shared/openOauthSignInWindow.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalInline.tsx`, `packages/thirdweb/src/wallets/in-app/web/lib/auth/index.ts`, `packages/thirdweb/src/wallets/in-app/web/lib/get-url-token.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Details.tsx`, `packages/thirdweb/src/react/web/wallets/shared/SocialLogin.tsx`, `packages/thirdweb/src/wallets/in-app/core/wallet/index.ts`, `packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx`, `packages/thirdweb/src/react/core/hooks/wallets/useAutoConnect.ts`, `packages/thirdweb/src/wallets/in-app/web/lib/web-connector.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/WalletSelector.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
* **#3809** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts [hunk 1]
```diff
@@ -783,4 +783,9 @@ export type ConnectButtonProps = {
    * Refer to the [`SiweAuthOptions`](https://portal.thirdweb.com/references/typescript/v5/SiweAuthOptions) for more details
    */
   auth?: SiweAuthOptions;
+
+  /**
+   * Set to true to use a redirect when logging in via OAuth instead of a pop-up window
+   */
+  socialLoginRedirect?: boolean;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts:790]: Prob better as an optional union type inside 'auth' like `mode: "popup" | "redirect"`

File: packages/thirdweb/src/utils/domains.ts [hunk 2]
```diff
@@ -26,11 +26,11 @@ type DomainOverrides = {
...
-export const DEFAULT_RPC_URL = "rpc.thirdweb.com";
-const DEFAULT_IN_APP_WALLET_URL = "embedded-wallet.thirdweb.com";
-const DEFAULT_PAY_URL = "pay.thirdweb.com";
-const DEFAULT_STORAGE_URL = "storage.thirdweb.com";
-const DEFAULT_BUNDLER_URL = "bundler.thirdweb.com";
+export const DEFAULT_RPC_URL = "rpc.thirdweb-dev.com";
+const DEFAULT_IN_APP_WALLET_URL = "embedded-wallet.thirdweb-dev.com";
+const DEFAULT_PAY_URL = "pay.thirdweb-dev.com";
+const DEFAULT_STORAGE_URL = "storage.thirdweb-dev.com";
+const DEFAULT_BUNDLER_URL = "bundler.thirdweb-dev.com";
```
Comment (joaquim-verges) [inline packages/thirdweb/src/utils/domains.ts:33]: Remember to revert

File: packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts [hunk 3]
```diff
@@ -783,4 +783,9 @@ export type ConnectButtonProps = {
    * Refer to the [`SiweAuthOptions`](https://portal.thirdweb.com/references/typescript/v5/SiweAuthOptions) for more details
    */
   auth?: SiweAuthOptions;
+
+  /**
+   * How the OAuth login page opens, defaults to 'popup'
+   */
+  socialLoginMode?: "popup" | "redirect";
```
Comment (MananTank) [inline packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts:790]: Shouldn't this be an option of `inAppWallet()` instead?

Comment (joaquim-verges) [reply packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts:790]: yes @gregfromstl this needs to be inside the inAppWallet props i think, would put it at the same level as where you choose your oauth login options

**Pull request closed without merging**